lib/Github/Api/GitData/References.php

```diff
@@ -25,6 +25,34 @@ public function all($username, $repository)
         return $this->get('/repos/'.rawurlencode($username).'/'.rawurlencode($repository).'/git/refs');
     }
 
+    /**
+     * Get all matching references for a particular branch of a repository.
+     *
+     * @param string $username
+     * @param string $repository
+     * @param string $branch
+     *
+     * @return array
+     */
+    public function matchingBranch($username, $repository, $branch)
+    {
+        return $this->get('/repos/'.rawurlencode($username).'/'.rawurlencode($repository).'/git/matching-refs/heads/'.$branch);
+    }
+
+    /**
+     * Get all matching references for a particular branch of a repository.
+     *
+     * @param string $username
+     * @param string $repository
+     * @param string $tag
+     *
+     * @return array
+     */
+    public function matchingTag($username, $repository, $tag)
```

Comment:
As this is new code I would actually add the `string` typehints to the parameters

Reply:
Excellent. I wasn't quite sure what version of PHP to target, so I stuck with what the other methods were doing and not type hint.  Will add them back in.